pkg/apis/core/validation/validation_test.go

```diff
@@ -16948,6 +16964,81 @@ func fmtErrs(errs field.ErrorList) string {
 	return buf.String()
 }
 
+func verifyVersionedValidationEquivalence(t *testing.T, obj, old k8sruntime.Object) {
+	t.Helper()
+
+	// Accumulate errors from all versioned validation, per version.
+	all := map[string]field.ErrorList{}
+	accumulate := func(t *testing.T, gv string, errs field.ErrorList) {
+		all[gv] = errs
+	}
+	if old == nil {
+		runtimetest.RunValidationForEachVersion(t, legacyscheme.Scheme, sets.Set[string]{}, obj, accumulate)
```

Comment:
To better support the pattern being established in this PR, I'm okay with replacing `RunValidationForEachVersion`  with a function that returns a `map[string]field.ErrorList{}` directly.

Reply:
I think this is OK as it is - it's not terrible, and it is contained in this one func.  I will go witth whatever you think gets past deads2k :)